sdn_tests/testplans/Stress_Latency_Test_Plan.md

```diff
@@ -0,0 +1,385 @@
+
+# E2E Stress Test Cases
+
+The purpose of the gNMI stress test is to validate the robustness of the telemetry service and overall performance. The stress testing will not validate the correctness of the data that is being received from the server.  Since the gNMI performance is dependent on the platform, the tests will not enforce the threshold to be validated as part of the validation, instead the tests will gather the metrics and produce the performance report at the end of the tests. The final report will be analyzed and checked against the platform specific requirements. There will also be periodic sanity checks during the test to capture test failures early.
+
+The robustness is validated through the following methods
+
+-   Monitoring the CPU load for the gNMI server
+-   Validation for the any unwanted process termination during the test
+-   Monitoring the memory usage of the system during the test
+-   Validate that the responses are received in the client from the gNMI server.
+
+## SET UPDATE operation
+
+### Expectation
+
+The expectation is for the gNMI server to process the valid update request and populate the leaf nodes with the values specified in the request payload and return the valid response to the client. If the leaf nodes are not present in the payload then the server should not populate them with the default values and leave those leaf values as-is(not modify them). The gNMI specification description can be found [here](https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-specification.md#344-modes-of-update-replace-versus-update).
+
+## SET REPLACE operation
+
+### Expectation
+
+The expectation is for the gNMI server to process the valid replace request and populate the leaf nodes with the values specified in the request payload and return the valid response to the client. If the leaf nodes are not present in the payload then the server should populate them with the default values. The gNMI specification description can be found [here](https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-specification.md#344-modes-of-update-replace-versus-update).
+
+## SET DELETE operation
+
+### Expectation
+
+The expectation is for the gNMI server to process the valid delete request and remove those nodes from the system. The server should send a valid response to the client once the request has been processed successfully.
+
+## GET operation
+
+### Expectation
+
+The GET response should contain the value of the request nodes. The expectation is for the gNMI server to process the valid get request and should send a valid response containing the values of the request nodes. Only the PROTO and JSON encodings are currently supported in the GPINS.
```

Comment:
The last sentence here can be removed ("Only the PROTO and JSON encodings...").

Reply:
Done.